skywalking/plugins/sw_loguru.py

```diff
@@ -0,0 +1,224 @@
+                status = core.activation_none
+                enabled[name] = status
+                if not status:
+                    return
+            else:
+                dotted_name = name + '.'
+                for dotted_module_name, status in core.activation_list:
+                    if dotted_name[: len(dotted_module_name)] == dotted_module_name:
+                        if status:
+                            break
+                        enabled[name] = False
+                        return
+                enabled[name] = True
+
+        current_datetime = aware_now()
+
+        if level_id is None:
+            level_icon = ' '
+            level_no = static_level_no
+            level_name = f'Level {level_no}'  # not really level name, just as loguru
+        else:
+            level_name, level_no, _, level_icon = core.levels[level_id]
+
+        if level_no < core.min_level:
+            return
+
+        code = frame.f_code
+        file_path = code.co_filename
+        file_name = basename(file_path)
+        thread = current_thread()
+        process = current_process()
+        elapsed = current_datetime - start_time
+
+        if exception:
+            if isinstance(exception, BaseException):
+                type_, value, traceback = (type(exception), exception, exception.__traceback__)
+            elif isinstance(exception, tuple):
+                type_, value, traceback = exception
+            else:
+                type_, value, traceback = sys.exc_info()
+            exception = RecordException(type_, value, traceback)
+        else:
+            exception = None
+
+        log_record = {
+            'elapsed': elapsed,
+            'exception': exception,
+            'extra': {**core.extra, **logger_context.get(), **extra},
+            'file': RecordFile(file_name, file_path),
+            'function': code.co_name,
+            'level': RecordLevel(level_name, level_no, level_icon),
+            'line': frame.f_lineno,
+            'message': str(message),
+            'module': splitext(file_name)[0],
+            'name': name,
+            'process': RecordProcess(process.ident, process.name),
+            'thread': RecordThread(thread.ident, thread.name),
+            'time': current_datetime,
+        }
+
+        if capture and kwargs:
+            log_record['extra'].update(kwargs)
+
+        if record:
+            kwargs.update(record=log_record)
+
+        if args or kwargs:
+            log_record['message'] = message.format(*args, **kwargs)
+
+        if core.patcher:
+            core.patcher(log_record)
+
+        if patcher:
+            patcher(log_record)
+
+        return log_record
+
+    def _sw_log(self, level_id, static_level_no, from_decorator, options, message, args, kwargs):
+        _log(level_id, static_level_no, from_decorator, options, message, args, kwargs)
+        record = gen_record(self, level_id, static_level_no, from_decorator, options, message, args, kwargs)
```

Comment:
NoneType bug here, your gen_record function can return None if core handler missing (from `logger.remove()`), then it will error line 156

Reply:
> NoneType bug here, your gen_record function can return None if core handler missing (from `logger.remove()`), then it will error line 156

yeah, so carefully 👍